.azure-pipelines/pull_request.yml

```diff
@@ -47,6 +49,28 @@ stages:
              find . -not -path "*/\.*" -type f -name "*.ipynb"\
               | xargs jupyter nbconvert --to python --output-dir tests\
               --RegexRemovePreprocessor.patterns="['^# exclude-from-export']"
-             python tests/bragg-edge-imaging-2D.py
-             python tests/sans-reduction.py
-
+             touch $(test_output)
+             echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?><testsuites><testsuite name=\"notebooks\">" >> $(test_output)
+             exclude_list=(dataconfig.py)
+             for f in tests/*.py; do 
+                 found=0
+                 for g in "${exclude_list[@]}" ; do
+                     [ "$f" = "tests/$g" ] && found=1
+                 done
+                 if [ "$found" = 0 ]; then
+                     python "$f"
+                     if [ $? != 0 ]; then
+                         echo "<testcase classname=\"class\" name=\"$f\"><failure message=\"$f failed\"/></testcase>" >> $(test_output)
+                     else
+                         echo "<testcase classname=\"class\" name=\"$f\"/>" >> $(test_output)
+                     fi
+                 fi
+             done
```

Comment:
Looks good. Could this be moved to a common template? https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#insert-a-template

Reply:
ah, template is the search term I needed. The thought occurred, but I looked for "importing YAML" and got nowhere.
I'll try this.